src/manager.ts

```diff
@@ -32,6 +60,20 @@ class ContextManager implements IDisposable {
     if (this.isDisposed) {
       return;
     }
-    this._context = null;
+    this.clearManager()
+    this._context = null
+  }
+
+  clearManager() {
```

Comment:
I think this should all live inside of the ``dispose`` method, but things aren't being cleaned up correctly. It might be because we're attaching object refs to window?...

Reply:
Definitely something odd going on, it seemed to be working in my previous approach, but isolating it on the dispose method is certainly cleaner. When is the ``ContextManager.dispose`` actually called?

Reply:
Previously I was cleaning up whenever the kernel was restarted, dispose is now only called when the notebook tab is closed. Should it be called for both?